src/index.ts

```diff
@@ -348,12 +359,254 @@ async function certificateForImpl<
+ * @public
+ * @param hostname - hostname of the remote machine
+ * @param certPath - file path to store the cert
+ * @param param2 - TrustRemoteOptions options
+ */
+export async function trustRemoteMachine(
+  hostname: string,
+  certPath: string,
+  {
+    port = DEFAULT_REMOTE_PORT,
+    renewalBufferInBusinessDays = REMAINING_BUSINESS_DAYS_VALIDITY_BEFORE_RENEW,
+    logger
+  }: Partial<TrustRemoteOptions> = {}
+): Promise<{ mustRenew: boolean }> {
+  debug('fetching/generating domain cert data for connecting to remote');
+  const returnInfo = new Deferred<{ mustRenew: boolean }>();
+  const { cert, key } = await certificateFor(
+    'devcert-domain-cert',
+    [hostname],
+    {
+      skipHostsFile: true
+    }
+  );
+  const certData = cert.toString();
+  const keyData = key.toString();
+  _logOrDebug(logger, 'log', `Connecting to remote host ${hostname} via ssh`);
+  // Connect to remote box via ssh.
+  const child = execa.shell(
+    // @TODO Change this to npx
+    `ssh ${hostname} node devcert/bin/devcert.js remote --port=${port} --cert='${JSON.stringify(
+      certData
+    )}' --key='${JSON.stringify(keyData)}'`,
+    {
+      detached: false
+    }
+  );
+
+  // Error handling for missing handles on child process.
+  if (!child.stderr) {
+    throw new Error('Missing stderr on child process');
+  }
+  if (!child.stdout) {
+    throw new Error('Missing stdout on child process');
+  }
+
+  // Throw any error that might have occurred on the remote side.
+  child.stderr.on('data', (data: execa.StdIOOption) => {
+    if (data) {
+      const stdErrData = data.toString().trimRight();
+      debug(stdErrData);
+      if (stdErrData.toLowerCase().includes('error')) {
+        debug('Error thrown on the remote side. Closing Remote server');
+        closeRemoteServer(hostname, port);
+        throw new Error(stdErrData);
+      }
+    } else {
+      debug('Stderr: {}');
+    }
+  });
+
+  // Listen to the stdout stream and determine the appropriate steps.
+  _logOrDebug(
+    logger,
+    'log',
+    `Attempting to start the server at port ${port}. This may take a while...`
+  );
+  child.stdout.on('data', (data: execa.StdIOOption) => {
+    if (data) {
+      const stdoutData = data.toString().trimRight();
+      if (stdoutData.includes(`STATE: READY_FOR_CONNECTION`)) {
+        _logOrDebug(
+          logger,
+          'log',
+          `Connected to remote host ${hostname} via ssh successfully`
+        );
+        // Once certs are trusted, close the remote server and cleanup.
+        _trustRemoteMachine(hostname, certPath, {
+          port,
+          renewalBufferInBusinessDays,
+          logger
+        })
+          .then(mustRenew => {
+            debug(
+              `Certs trusted successfully, the value of mustRenew is ${mustRenew}`
+            );
+            // return the certificate renewal state to the consumer to handle the
+            // renewal usecase.
+            child.kill();
```

Comment:
now I'm wondering if we ever need to kill the child process at all in this case. Doesn't this mean that it's successfully completed (i.e., `exit 0`?)

Reply:
yep... we shouldnt need this anymore. I will double check, but pretty sure about that